main.tf

```
@@ -1,3 +1,324 @@
+  #     }
+  #   }
+  # }
+  configs = {
+    for key, value in module.deep : key => value.merged
+  }
+
+  # Get the Stacks configs, this is just to improve code readability
+  # Example:
+  # {
+  #   "random-pet-example" = {
+  #     "manage_state" = true
+  #   }
+  # }
+  stack_configs = {
+    for key, value in local.configs : key => value.stack_settings
+  }
+
+  # Get the list of all stack names
+  stacks = toset(keys(local.stack_configs))
+
+  ## Labels
+  # Сreates a map of administrative labels for each stack that has the administrative property set to true.
+  # Example:
+  # {
+  #   "spacelift-automation-mp-main" = [
+  #     "administrative",
+  #   ]
+  #   "spacelift-policies-notify-tf-completed" = [
+  #     "administrative",
+  #   ]
+  # }
+
+  _administrative_labels = {
+    for stack, configs in local.stack_configs : stack => ["administrative"] if tobool(try(configs.administrative, false)) == true
+  }
+
+  # Creates a map of `depends-on` labels for each stack based on the root module level dependency configuration.
+  # Example:
+  # {
+  #   "random-pet-example" = [
+  #     "depends-on:spacelift-automation-default",
+  #   ]
+  # }
+
+  _dependency_labels = {
+    for stack in local.stacks : stack => [
+      "depends-on:spacelift-automation-${terraform.workspace}"
+    ]
+  }
+
+  # Creates a map of folder labels for each stack based on Git structure for a proper grouping stacks in Spacelift UI.
+  # https://docs.spacelift.io/concepts/stack/organizing-stacks#label-based-folders
+  # Example:
+  # {
+  #   "random-pet-example" = [
+  #     "folder:random-pet/example",
+  #   ]
+  # }
+
+  _folder_labels = {
+    for stack in local.stacks : stack => [
+      "folder:${local.configs[stack].root_module}/${local.configs[stack].terraform_workspace}"
+    ]
+  }
+
+  # Merge all the labels into a single map for each stack.
+  # Example:
+  # {
+  #   "random-pet-example" = tolist([
+  #     "folder:random-pet/example",
+  #     "depends-on:spacelift-automation-default",
+  #   ])
+  # }
+
+  labels = {
+    for stack in local.stacks :
+    stack => compact(flatten([
+      lookup(local._administrative_labels, stack, []),
+      lookup(local._folder_labels, stack, []),
+      lookup(local._dependency_labels, stack, []),
+      try(local.stack_configs[stack].labels, []),
+    ]))
+  }
+
+  # Merge all before_init steps into a single map for each stack.
+  before_init = {
+    for stack in local.stacks : stack => compact(concat(
+      var.before_init,
+      try(local.stack_configs[stack].before_init, []),
+      # This command is required for each stack.
+      # It copies the tfvars file from the stack's workspace to the root module's directory
+      # and renames it to `spacelift.auto.tfvars` to automatically load variable definitions for each run/task.
+      ["cp tfvars/${local.configs[stack].terraform_workspace}.tfvars spacelift.auto.tfvars"],
+    )) if try(local.configs[stack].tfvars.enabled, false)
```

I feel like we should default to tfvars being enabled... Or we should be very explicit about tfvars needing to be enabled for the tfvars file to get picked up by Spacelift. Do we do the latter at least? Let me check the readme...

---

See my update above to the README that goes with this. Let's make this default to `true`. 

```suggestion
    )) if try(local.configs[stack].tfvars.enabled, true)
```